Database.py

```diff
@@ -0,0 +1,78 @@
+from datetime import datetime
+from xmlrpc.client import Boolean
+from Employee import INVALID_DATETIME, INVALID_STR, Employee
+from EmployeeContainer import EmployeeContainer, EmployeeAdmin, EmployeeOther, EmployeeSelf
+from pathlib import Path
+import csv
+from tkinter import messagebox
+
+
+class Database:
+    def __init__(self, initFPath: Path = Path("")) -> None:
+        self.employeeList: list[Employee] = list()
+        if initFPath == "":
+            return
+        elif initFPath.is_file():
+            with initFPath.open() as csvfile:
+                csvReader = csv.DictReader(csvfile)
+                for row in csvReader:
+                    # test what happens if there are different fields from Employee class
+                    self.employeeList.append(Employee(**row))
+        elif initFPath != "":
+            print("not a filepath\n")
+
+    # do we need to do any validity checks before adding?
+    def addEmployee(self, **kwargs) -> Boolean:
+        if len(self.search(Employee_ID=kwargs["Emp_ID"])) > 0:
+            return False
+        self.employeeList.append(Employee(**kwargs))
+        return True
+
+    def search(self, Department: str = INVALID_STR, fName: str = INVALID_STR, lName: str = INVALID_STR, Employee_ID: str = INVALID_STR, Title: str = INVALID_STR, oPhone: str = INVALID_STR, StartDate: datetime = INVALID_DATETIME, EndDate: datetime = INVALID_DATETIME) -> list:
+        foundList = list()
+        for emp in self.employeeList:
+            # when user sessions are implemented, this needs to be replaced with a proper switch
+            emp = EmployeeOther(emp)
+            if (Department == INVALID_STR or Department == emp.Dept) and (fName == INVALID_STR or fName == emp.getFName()) and (lName == INVALID_STR or lName == emp.getLName()) and (Employee_ID == INVALID_STR or Employee_ID == emp.EmpID) and (Title == INVALID_STR or Title == emp.Title) and (oPhone == INVALID_STR or oPhone == emp.OfficePhone) and (StartDate == INVALID_DATETIME or StartDate == emp.StartDate) and (EndDate == INVALID_DATETIME or EndDate == emp.EndDate):
+                foundList.append(emp)
+        return foundList
+
+    def importDB(self, importFPath: Path = Path("")) -> None:
+        if importFPath.is_file():
+            impList: list[Employee] = list()
+            with importFPath.open() as csvfile:
+                csvReader = csv.DictReader(csvfile)
+                for row in csvReader:
+                    # test what happens if there are different fields from Employee class
+                    impList.append(Employee(**row))
+            for emp in impList:
+                dupeList: list[Employee] = list()
```

Comment:
dupeList needs to be reset every iteration. it is a list of all stored duplicates of the specific employee. we don't treat it as a list, instead as just an employee, because there shouldnt be more than one duplicate, as if there were two, then they would be duplicates of eachother. it being a list is just in case we decide to add a ffeature for handling that.

Reply:
I understand that and that's why I'm saying still reset it, but instead of initializing it all over again you can just create it once outside the for loop, and after you take care of the overwrite you empty the list, instead of needing to reinitialize it each time.

Reply:
that would change its scope. it would still work, but generally the more local the variable's scope the better. it's not like creating an empty list is performance heavy either.

Reply:
ok, it's not a big deal, I'm fine keeping it